### PR TITLE
[Do-not-merge] Upgrade to Cosmos-SDK v0.50.

### PR DESCRIPTION
**Pull request by alpe**
🚧 WIP - early version

See #1438

⚠️ All broken currently ! 🤯 

Failing tests are deactivated. Go mod is setup my local workspace. 

I am waiting for a soution in ibc-go that fixes the: `failed consensus state verification for client (07-tendermint-0): chained membership proof failed to verify membership of value:`

See [upgrade doc](https://github.com/cosmos/cosmos-sdk/blob/release/v0.50.x/UPGRADING.md) in the SDK

I have started with just the go.mod dependencies and some related changes to imports.

This blocked now on the ibc-go release that includes the `capability` module and fixed imports for sdk v0.50. 
* Notional is helping the team with a [PR](https://github.com/cosmos/ibc-go/pull/3883) but still core integration of the testing framework does not work for me.
* Either there are some issues with CometBFT or the way how I would use it for the extended ibc-go testing framework. See my [issue](https://github.com/cometbft/cometbft/issues/1001)

This is also blocked on on some modules in the SDK that are not released. See `go.mod` replacement statements for details

Some notes on the work

#### New modules
- circuit

#### Moved to `cosmossdk.io` path
- store
- x/nft
- x/evidence
- x/feegrant
- x/upgrade

#### Others
"github.com/cosmos/cosmos-sdk/snapshots -> "cosmossdk.io/store/snapshots
"github.com/cosmos/cosmos-sdk/client/grpc/tmservice -> "github.com/cosmos/cosmos-sdk/client/grpc/cmtservice
"github.com/cosmos/cosmos-sdk/x/capability -> "github.com/cosmos/ibc-go/v7/modules/capability

Removed: legacy proposals for x/upgrade
### 
- [x] deprecate MakeEncodingConfig
- [x] wasm keeper: accept runtime.NewKVStoreService for keys[wasm.StoreKey],
- [x] handle storeservice.Get/Set errors 
- [ ] remove non requires sdk modules from begin/ endblocker 
- [ ] replace sdk.Context with stdlib context in keeper
- [x] use `[(cosmos_proto.scalar) = "cosmos.AddressString"]` for addresses in proto
- [x] set `cosmos.msg.v1.service` proto annotation
- [x] upgrade ante-handler
- [x] use new collections framework for params

### CODE REVIEW

File: app/app.go [hunk 1]
```diff
@@ -780,8 +741,12 @@ func NewWasmApp(
 			govtypes.ModuleName: gov.NewAppModuleBasic(
 				[]govclient.ProposalHandler{
 					paramsclient.ProposalHandler,
+					// non sdk handlers
+					ibcclientclient.UpdateClientProposalHandler,
+					ibcclientclient.UpgradeProposalHandler,
 				},
 			),
+			ibctm.ModuleName: ibctm.AppModuleBasic{},
```
Comment (sontrinh16) [inline app/app.go:748]: i did have some problem with the unrecognize url type related to ibc light client tendermint types. Dont know if you solve it with this line but i have to add an extra line to register the interface like this `	ibctm.RegisterInterfaces(interfaceRegistry)
` below

Comment (alpe) [reply app/app.go:748]: @sontrinh16 I had the same problem but this line fixes the issue.

But this only brings me to the next problem which is `failed consensus state verification for client (07-tendermint-0): chained membership proof failed to verify membership of value:` . If you have any solution to this, that would be super helpful!

**Comment by faddat** [comment]
@alpe might be able to make a pr to fix now.

**Comment by faddat** [comment]
@alpe sounds good to me. 

FYI:

1) I am not sure if rebase is best, typically I fail with that. 

2) If you are sure that it is best, would you mind showing me how you do it, when you do it?  Because if it is best, I'm knowledge deficient there.

**Comment by alpe** [comment]
Closing this in favour of https://github.com/CosmWasm/wasmd/pull/1528

**Pull request closed without merging**